src/websocket_service/handlers/push_delete.rs

```diff
@@ -26,90 +28,61 @@ pub async fn handle(
     let database = &state.database;
     let subscription_id = params.id;
 
-    match database
+    let Ok(Some(LookupEntry {
+        project_id, 
+        account,
+        ..
+    }))= database
         .collection::<LookupEntry>("lookup_table")
-        .find_one_and_delete(doc! {"_id": &topic.clone().to_string() }, None)
+        .find_one_and_delete(doc! {"_id": &topic.to_string() }, None)
         .await
-    {
-        Ok(Some(LookupEntry {
-            project_id,
-            account,
-            ..
-        })) => {
-            match database
-                .collection::<ClientData>(&project_id)
+        else {
+            return Err(Error::NoProjectDataForTopic(topic.to_string()))
+        };
+
+    let Ok(Some(acc)) = database
+                 .collection::<ClientData>(&project_id)
                 .find_one_and_delete(doc! {"_id": &account }, None)
-                .await
-            {
-                Ok(Some(acc)) => {
-                    match base64::engine::general_purpose::STANDARD
-                        .decode(params.data.message.to_string())
-                    {
-                        Ok(message_bytes) => {
-                            let envelope = EnvelopeType0::from_bytes(message_bytes);
-                            // Safe unwrap - we are sure that stored keys are valid
-                            let encryption_key = hex::decode(&acc.sym_key)?;
-                            let cipher =
-                                ChaCha20Poly1305::new(GenericArray::from_slice(&encryption_key));
+                .await else {
+                    return Err(Error::NoClientDataForTopic(topic.to_string()))
+                };
+
+    let Ok(message_bytes) = base64::engine::general_purpose::STANDARD
+        .decode(params.data.message.to_string()) else {
+            Err(anyhow!("Failed to decode message"))?
```

Comment:
`?` is being desugared into `match { Ok, Err }` When you look at it this way such `?` usage may make you feel some itch :smile: 

`return` will do just fine